packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -26,31 +30,34 @@ class _SimpleExpansionPanelListTestWidgetState extends State<SimpleExpansionPane
 
   @override
   Widget build(BuildContext context) {
-    return ExpansionPanelList(
-      expansionCallback: (int _index, bool _isExpanded) {
-        setState(() {
-          extendedState[_index] = !extendedState[_index];
-        });
-      },
-      children: <ExpansionPanel>[
-        ExpansionPanel(
-          headerBuilder: (BuildContext context, bool isExpanded) {
-            return Text(isExpanded ? 'B' : 'A', key: widget.firstPanelKey);
-          },
-          body: const SizedBox(height: 100.0),
-          canTapOnHeader: widget.canTapOnHeader,
-          isExpanded: extendedState[0],
-        ),
-        ExpansionPanel(
-          headerBuilder: (BuildContext context, bool isExpanded) {
-            return Text(isExpanded ? 'D' : 'C', key: widget.secondPanelKey);
-          },
-          body: const SizedBox(height: 100.0),
-          canTapOnHeader: widget.canTapOnHeader,
-          isExpanded: extendedState[1],
-        ),
-      ],
-    );
+    final ExpansionPanelCallback expansionPanelCallback = (int _index, bool _isExpanded) {
+      setState(() {
+        extendedState[_index] = !extendedState[_index];
+      });
+    };
+
+    final List<ExpansionPanel> children = <ExpansionPanel>[
+      ExpansionPanel(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'B' : 'A', key: widget.firstPanelKey);
+        },
+        body: const SizedBox(height: 100.0),
+        canTapOnHeader: widget.canTapOnHeader,
+        isExpanded: extendedState[0],
+      ),
+      ExpansionPanel(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'D' : 'C', key: widget.secondPanelKey);
+        },
+        body: const SizedBox(height: 100.0),
+        canTapOnHeader: widget.canTapOnHeader,
+        isExpanded: extendedState[1],
+      ),
+    ];
+
+    return widget.expandedHeaderPadding == null
+        ? ExpansionPanelList(expansionCallback: expansionPanelCallback, children: children)
+        : ExpansionPanelList(expansionCallback: expansionPanelCallback, children: children, expandedHeaderPadding: widget.expandedHeaderPadding);
```

Comment:
I think this refactor can be pretty confusing for developers looking at this for the first time (they'd have to dig into `expansion_panel.dart` and figure out why this was written this way).

Reply:
You are right it's quite confusing. Please have a look at the latest draft.